src/Bundle/test/src/Mailer/FileTransport.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace App\Mailer;
+
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Mime\RawMessage;
+
+final class FileTransport implements TransportInterface
+{
+    public function __construct(private string $spoolDirectory)
+    {
+    }
+
+    public function send(RawMessage $message, Envelope $envelope = null): ?SentMessage
+    {
+        if (!is_dir($this->spoolDirectory)) {
+            mkdir($this->spoolDirectory);
+        }
+
+        $filename = (string) rand(10000, 1000000) . '.message';
```

Comment:
Why do we rand number here? I guess numbers are so large to not care about naming collisions? Won't it be better to use something like https://www.php.net/manual/en/function.tempnam.php?